### PR TITLE
A somewhat hacky way of making circle link against clang trunk's libs

### PR DESCRIPTION
**Pull request by mattgodbolt**
Not glorious but seems to work locally.

CC @seanbaxter

### CODE REVIEW

**Comment by partouf** [comment]
can you use latest stableish clang pls, libraries are complicated enough as is

**Comment by mattgodbolt** [comment]
I made it configurable and pointed it at latest clang :) thanks for the suggestion

File: lib/buildenvsetup/ceconan-circle.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
```
Comment (partouf) [inline lib/buildenvsetup/ceconan-circle.ts:25]: don't need any of these top imports as far as I can see

File: lib/buildenvsetup/ceconan-circle.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        super(compilerInfo, env);
+
+        this.linkedCompilerId = compilerInfo.buildenvsetup.props('linkedCompilerId');
+    }
+
+    override async getConanBuildProperties(key): Promise<ConanBuildProperties> {
+        const props = await super.getConanBuildProperties(key);
+        props['compiler'] = 'clang';
+        props['compiler.version'] = this.linkedCompilerId;
+        props['arch'] = 'x86_64'; // or better still work out how best to get this in the base
```
Comment (partouf) [inline lib/buildenvsetup/ceconan-circle.ts:55]: I think you can use https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/buildenvsetup/base.ts#L154 by saying `props['arch'] = this.getTarget(key)`

Comment (mattgodbolt) [reply lib/buildenvsetup/ceconan-circle.ts:55]: That didn't work and I couldn't work out why or what was missing. It kept aertingnit to the empty string

Comment (mattgodbolt) [reply lib/buildenvsetup/ceconan-circle.ts:55]: Defaulting* sorry thumbs and phone is having bad time

Comment (partouf) [reply lib/buildenvsetup/ceconan-circle.ts:55]: ah :(
you might need to add an exception here https://github.com/compiler-explorer/compiler-explorer/blob/633eb82d188474a700b2dc8edca3f3c5e15797fc/lib/buildenvsetup/base.ts#L67 to return true if the this.compiler.exe is circle

Comment (mattgodbolt) [reply lib/buildenvsetup/ceconan-circle.ts:55]: That I can do

**Comment by partouf** [comment]
@mattgodbolt how have you tested this? did you test it with UB20?

I'm getting this error locally:

```
/usr/bin/ld: ./libfmtd.a(format.cc.o): in function `std::__new_allocator<unsigned int>::allocate(unsigned long, void const*)':
format.cc:(.text._ZNSt15__new_allocatorIjE8allocateEmPKv[_ZNSt15__new_allocatorIjE8allocateEmPKv]+0x4a): undefined reference to `std::__throw_bad_array_new_length()'
```

I don't get it when I'm using Clang 16 or 15 when trying to link with fmt

**Comment by partouf** [comment]
I couldn't get this to work with Clang because of linking errors, I changed it to match the gcc toolchain instead and now it works

**Comment by partouf** [comment]
this is now live

**Pull request merged**